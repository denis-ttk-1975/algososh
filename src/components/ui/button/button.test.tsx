import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './button';

it('компонент Button без пропсов', () => {
  const element = renderer.create(<Button />).toJSON();
  expect(element).toMatchSnapshot();
});

it('компонент Button c пропсом text', () => {
  const element = renderer.create(<Button text='Several words' />).toJSON();
  expect(element).toMatchSnapshot();
});

it('компонент Button c пропсом disabled', () => {
  const element = renderer.create(<Button disabled={true} />).toJSON();
  expect(element).toMatchSnapshot();
});

it('компонент Button c пропсом isLoader', () => {
  const element = renderer.create(<Button isLoader={true} />).toJSON();
  expect(element).toMatchSnapshot();
});

it('Нажатие на кнопку вызывает корректный коллбэк', () => {
  window.alert = jest.fn();

  render(<Button text='Callback ignition test' onClick={() => alert('Button pressed')} />);

  const button = screen.getByText('Callback ignition test');

  fireEvent.click(button);

  expect(window.alert).toHaveBeenCalledWith('Button pressed');
});
