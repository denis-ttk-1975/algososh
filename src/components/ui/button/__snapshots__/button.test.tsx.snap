// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`компонент Button c пропсом disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`компонент Button c пропсом isLoader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`компонент Button c пропсом text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Several words
  </p>
</button>
`;

exports[`компонент Button без пропсов 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
